tests/test_visual.py

```diff
@@ -101,21 +107,45 @@ def test_plot_bboxes():
     img = Image.open(
         str(cur_dir / "mock_data" / "simrun" / "captures" / "camera_000.png")
     )
+    label_mappings = {1: "car", 2: "tree", 3: "light"}
     boxes = [
-        BBox2D(label="car", x=1, y=1, w=2, h=3),
-        BBox2D(label="tree", x=7, y=6, w=3, h=4),
-        BBox2D(label="light", x=2, y=6, w=2, h=4),
-    ]
-    colors = [
-        ImageColor.getcolor("green", "RGB"),
-        ImageColor.getcolor("red", "RGB"),
-        ImageColor.getcolor("green", "RGB"),
+        BBox2D(label=1, x=1, y=1, w=2, h=3),
+        BBox2D(label=1, x=7, y=6, w=3, h=4),
+        BBox2D(label=1, x=2, y=6, w=2, h=4),
     ]
+    colors = ["green", "red", "green"]
 
     with patch(
-        "datasetinsights.stats.visualization.plots.ImageDraw.Draw"
+        "datasetinsights.stats.visualization.plots.add_single_bbox_on_image"
     ) as mock:
-        instance = mock.return_value
-        plot_bboxes(img, boxes, colors)
-        assert instance.rectangle.call_count == len(boxes)
-        assert instance.multiline_text.call_count == len(boxes)
+        plot_bboxes(img, boxes, label_mappings=label_mappings, colors=colors)
+        assert mock.call_count == len(boxes)
+
+
+@patch("datasetinsights.evaluation_metrics.confusion_matrix.Records")
+def test_match_boxes(mock_record):
+    match_results = [(0.5, True), (0.6, False), (0.7, True)]
+    expected_colors = ["green", "red", "green"]
+    mock_record.return_value.match_results.return_value = match_results
+    colors = match_boxes(None, None)
+    for i in range(len(colors)):
+        assert colors[i] == expected_colors[i]
+
+
+@patch("datasetinsights.stats.visualization.bbox2d_plot._cv2.rectangle")
+def test__add_single_bbox_on_image(mock):
```

Comment:
This test does two things. A test should follow the structure arrange/act/assert. 
This test has the following structure arrange/act/assert/act/arrange. This means that it tests multiple things in one test. The test should be refactored to separate the cases.
Secondly while its good to check that the mocks are called, they don't validate the behavior meaning what are the parameters the mocks are called with. This is the important part of the behavior and should be tested.

Reply:
Separate the cases and mocked the parameters with `called_with`.